# HG changeset patch
# User akira noda <you.akira.noda@gmail.com>
# Date 1440176752 14400
#      Fri Aug 21 13:05:52 2015 -0400
# Node ID 03a20edfe1c7f147e7adcface83025351b8f31c9
# Parent  0fe7133da8ceb02744b3288841927f586b9a514c
Fix compilation of classdef with the clang compiler (bug #41178)

* ov-classdef.cc, ov-classdef.h (cdef_object_scalar::mark_as_constructed):
Move function definition out of header file to avoid dependency on incomplete
type, compilation error with clang.

diff --git a/libinterp/octave-value/ov-classdef.cc b/libinterp/octave-value/ov-classdef.cc
--- a/libinterp/octave-value/ov-classdef.cc
+++ libinterp/octave-value/ov-classdef.cc
@@ -1946,6 +1946,12 @@ cdef_object_scalar::is_partially_constru
   return true;
 }
 
+void
+cdef_object_scalar::mark_as_constructed (const cdef_class& cls)
+{
+  ctor_list.erase (cls);
+}
+
 handle_cdef_object::~handle_cdef_object (void)
 {
 #if DEBUG_TRACE
diff --git a/libinterp/octave-value/ov-classdef.h b/libinterp/octave-value/ov-classdef.h
--- a/libinterp/octave-value/ov-classdef.h
+++ libinterp/octave-value/ov-classdef.h
@@ -438,7 +438,7 @@ public:
 
   void mark_as_constructed (void) { ctor_list.clear (); }
 
-  void mark_as_constructed (const cdef_class& cls) { ctor_list.erase (cls); }
+  void mark_as_constructed (const cdef_class& cls);
 
   bool is_constructed (void) const { return ctor_list.empty (); }
 
